GliaWidgets/Sources/QuickLookBased/QuickLookBased.Live.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+import UIKit
+import QuickLook
+
+extension QuickLookBased.ThumbnailGenerator {
+    static func live(_ generator: QLThumbnailGenerator) -> Self {
+        return .init(
+            generateBestRepresentation: { request, completion in
+                generator.generateBestRepresentation(for: request) { thumbnail, error in
+                    completion(thumbnail.map(QuickLookBased.ThumbnailRepresentation.init(live:)), error)
+                }
+            }
+        )
+    }
+
+    func createImageThumbnail(
+        fileURL: URL,
+        size: CGSize,
+        scale: CGFloat,
+        completion: @escaping (UIImage?) -> Void
+    ) {
+        let newSize = CGSize(width: size.width * scale, height: size.height * scale)
+        let request = QLThumbnailGenerator.Request(
+            fileAt: fileURL.standardizedFileURL,
+            size: newSize,
+            scale: scale,
+            representationTypes: .thumbnail
+        )
+        generateBestRepresentation(request) { thumbnail, error in
+            if let image = thumbnail?.uiImage {
+                completion(image)
+            } else if error != nil {
+                // TODO: need to handle error state
```

Comment:
Maybe it makes sense to at least log error to console?

Reply:
abaeca442de15a713f0fe344a234883623185251